core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -174,15 +162,46 @@ oppia.directive('audioControls', [
               AudioPlayerService.getCurrentAudioControlsDirectiveId();
           };
 
+          /**
+           * Called when an audio file finishes loading.
+           * @param {string} audioFilename - Filename of the audio file that
+           *                                 finished loading.
+           */
+          var onFinishedLoadingAudio = function(audioFilename) {
+            var mostRecentlyRequestedAudioFilename =
+              AudioPreloaderService.getMostRecentlyRequestedAudioFilename();
+            if (audioRequestIsPending &&
+                audioFilename === mostRecentlyRequestedAudioFilename) {
+              audioRequestIsPending = false;
+              AudioPlayerService.load(audioFilename, directiveId)
+                .then(function() {
+                  AudioPlayerService.play();
+                });
+            }
+            $scope.audioLoadingIndicatorIsShown = false;
```

Comment:
Originally I thought to display 'loading audio...' whenever the audio was actually loading, but thinking over it I think a better approach is to only display this when the audio is loading and the user has requested to play it, otherwise if the 'loading audio...' is already showing and the user requests to play they won't get any visual change/feedback. Removed audioRequestIsPending.